packages/graphql/src/schema-model/annotation/AuthorizationAnnotation.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { Annotation } from "./Annotation";
+
+export const AuthorizationAnnotationArguments = {
+    filter: "filter",
+    validate: "validate",
+};
+
+export const AuthorizationFilterOperationRule = [
+    "READ",
+    "UPDATE",
+    "DELETE",
+    "CREATE_RELATIONSHIP",
+    "DELETE_RELATIONSHIP",
+] as const;
+
+export const AuthorizationValidateOperationRule = [
+    "READ",
+    "CREATE",
+    "UPDATE",
+    "DELETE",
+    "CREATE_RELATIONSHIP",
+    "DELETE_RELATIONSHIP",
+] as const;
+
+export type AuthorizationFilterOperation =
+    (typeof AuthorizationFilterOperationRule)[keyof typeof AuthorizationFilterOperationRule];
+
+export type AuthorizationValidateOperation =
+    (typeof AuthorizationValidateOperationRule)[keyof typeof AuthorizationValidateOperationRule];
```

Comment:
I don't completely understand what these types are supposed to be, may there be a  more concise or explicit way of defining these?